drydock_provisioner/drydock_client/client.py

```diff
@@ -14,83 +14,213 @@
+            raise errors.ClientError("Received a %d from GET URL: %s" % (resp.status_code, endpoint),
+                                        code=resp.status_code)
+        else:
+            return resp.json()
+        
+    def create_design(self, base_design=None):
+        """
+        Create a new design context for holding design parts
+
+        :param string base_design: String UUID of the base design to build this design upon
+        :return string: String UUID of the design ID
+        """
+        endpoint = '/designs'
+
+        if base_design is not None:
+            resp = self.__send_post(endpoint, data={'base_design_id': base_design})
+        else:
+            resp = self.__send_post(endpoint)
+
+        if resp.status_code != 201:
+            raise errors.ClientError("Received a %d from POST URL: %s" % (resp.status_code, endpoint),
+                                        code=resp.status_code)
+        else:
+            design = resp.json()
+            return design.get('id', None)
+
+    def get_part(self, design_id, kind, key, source='designed'):
+        """
+        Query the model definition of a design part
+
+        :param string design_id: The string UUID of the design context to query
+        :param string kind: The design part kind as defined in the Drydock design YAML schema
+        :param string key: The design part key, generally a name.
+        :param string source: The model source to return. 'designed' is as input, 'compiled' is after merging
+        :return: A dict of the design part
+        """
 
-    def create_design(self):
+        endpoint = "/designs/%s/parts/%s/%s" % (design_id, kind, key)
 
-    def get_parts(self, design_id):
+        resp = self.__send_get(endpoint, query={'source': source})
 
-    def get_part(self, design_id, kind, key):
+        if resp.status_code == 404:
+            raise errors.ClientError("%s %s in design %s not found" % (key, kind, design_id), code=404)
+        elif resp.status_code != 200:
+            raise errors.ClientError("Received a %d from GET URL: %s" % (resp.status_code, endpoint),
+                                        code=resp.status_code)
 
     def load_parts(self, design_id, yaml_string=None):
+        """
+        Load new design parts into a design context via YAML conforming to the Drydock design YAML schema
+
+        :param string design_id: String uuid design_id of the design context
+        :param string yaml_string: A single or multidoc YAML string to be ingested
+        :return: Dict of the parsed design parts
+        """
 
+        endpoint = "/designs/%s/parts" % (design_id)
+
+        resp = self.__send_post(endpoint, query={'ingester': 'yaml'}, body=yaml_string)
+
+        if resp.status_code == 400:
+           raise errors.ClientError("Invalid inputs: %s" % resp.text, code=resp.status_code)
+        elif resp.status_code == 500:
+            raise errors.ClientError("Server error: %s" % resp.text, code=resp.status_code)
```

Comment:
Again, I think ClientError should be reserved just for 400-related codes while a new error type called ServerError or ServerFaultError or whatever should be used for 500-related error codes.